src/frontend/fluent-core/components/MediaItemList.ts

```diff
@@ -0,0 +1,151 @@
+        0% { transform: rotate(0deg); }
+        100% { transform: rotate(360deg); }
+    }
+
+    ul#entries {
+        list-style-type: none;
+        overflow-y: scroll;
+        overflow-x: hidden;
+        padding: 0;
+        margin: 0;
+    }
+
+    ul#entries li {
+        padding: calc(var(--design-unit) * 1px);
+        border-top: calc(var(--stroke-width) * 1px) solid var(--neutral-stroke-divider-rest);
+        gap: calc(var(--design-unit) * 1px);
+        display: grid;
+        align-items: center;
+        grid-template-rows: min-content;
+        grid-template-columns: min-content 1fr min-content;
+    }
+
+    ul#entries li > div {
+        overflow: hidden;
+        white-space: nowrap;
+        text-overflow: ellipsis;
+    }
+
+    ul#entries li .controls {
+        visibility: hidden;
+        display: flex;
+    }
+
+    ul#entries li:hover {
+        background-color: var(--neutral-fill-hover);
+    }
+
+    ul#entries li:hover .controls {
+        visibility: visible;
+    }
+`;
+
+const listitem: ViewTemplate<IMediaContainer> = html`
+    <li>
+        <div><!-- <fluent-checkbox></fluent-checkbox> --></div>
+        <div>${model => model.Title}</div>
+        <div class="controls">
+            <fluent-button appearance="stealth" title="${() => S.Locale.Frontend_FluentCore_MediaItemList_PreviewButton_Description()}" @click=${(model, ctx) => ctx.parent.ShowPreview(model)}>${IconPreview}</fluent-button>
+            <fluent-button appearance="stealth" title="${() => S.Locale.Frontend_FluentCore_MediaItemList_DownloadButton_Description()}" @click=${(model, ctx) => ctx.parent.Download(model)}>${IconDownload}</fluent-button>
+        </div>
+    </li>
+`;
+
+const template: ViewTemplate<MediaItemList> = html`
+    <div id="header">
+        <div id="title">${() => S.Locale.Frontend_FluentCore_MediaItemList_Heading()}</div>
+        <div class="hint">${model => model.filtered?.length ?? '┄'}／${model => model.entries?.length ?? '┄'}</div>
+    </div>
+    <div id="searchcontrol">
+        <fluent-searchbox allowcase allowregex @predicate=${(model, ctx) => model.match = (ctx.event as CustomEvent<(text: string) => boolean>).detail}></fluent-searchbox>
+    </div>
+    <ul id="entries">
+        ${repeat(model => model.filtered, listitem)}
+    </ul>
+`;
+
+@customElement({ name: 'fluent-media-item-list', template, styles })
+export class MediaItemList extends FASTElement {
+
+    @observable entries: IMediaContainer[] = [];
+    entriesChanged() {
+        this.FilterEntries();
+    }
+    @observable match: (text: string) => boolean = () => true;
+    matchChanged() {
+        this.FilterEntries();
+    }
+    @observable filtered: IMediaContainer[] = [];
+    @observable updating = false;
+
+    public async FilterEntries() {
+        this.filtered = this.entries?.filter(entry => this.match(entry.Title)).slice(0, 250) /* TODO: virtual scrolling */ ?? [];
+    }
+
+    public async ShowPreview(entry: IMediaContainer) {
```

Comment:
Implement a basic preview, so a website developer can show images directly without downloading them first

Reply:
Done in eb184eb6 and following commits